File: 0443-string-compression/0443-string-compression.cpp
class Solution {
public:
    int compress(vector<char>& chars) {
        int l = chars.size();
        if(l==0){
            return 0;
        }
        //char c = chars[0];
        int co = 1;
        string s;
        int i,k = 0;
        for(i=1;i<l;i++){
            if(chars[i]==chars[i-1]){
                co++;
            }
            else{
                chars[k]=chars[i-1];
                cout << k << endl;
                k++;
                // s.push_back(c);
                // cout << c << " " << "h" << endl;
                if(co>1){
                    string ss = to_string(co);
                    for(int j=0;j<ss.size();j++){
                        chars[k] = ss[j];
                        cout << k << endl;
                        k++;
                    }
                   // s = s + ss;
                }
                //c = chars[i];
                co = 1;
            }
        }
        // s.push_back(c);
        // if(co>1){
        //     string ss = to_string(co);
        //     s = s + ss;
        // }
        chars[k] = chars[i-1];
        k++;
        if(co>1){
            string ss = to_string(co);
            for(int j=0;j<ss.size();j++){
                chars[k] = ss[j];
                cout << k << endl;
                k++;
            }
        }
        
        // for(int i=0;i<s.size();i++){
        //     chars[i] = s[i];
        // }
        // return s.size();
        return k;
    }
};

File: 0443-string-compression/0443-string-compression.java
class Solution {
    public int compress(char[] chars) {
        int idx = 0;
        int len = chars.length;

        if(len == 1){
            return 1;
        }

        int j=1;
        int cnt = 1;
        char ch = chars[0];

        //let j iterate => whenever there is difference let's modify the characters.
        while(j < len){
            if(chars[j] == ch){
                cnt++;
            }
            else{
                chars[idx++] = ch;
                if(cnt > 1){
                    String countStr = String.valueOf(cnt);
                    for (int i = 0; i < countStr.length(); i++) {
                        chars[idx++] = countStr.charAt(i);
                    }
                }
                cnt = 1;
                ch = chars[j];
            }
            j++;
        }

        chars[idx++] = ch;
        if(cnt > 1){
            String countStr = String.valueOf(cnt);
            for (int i = 0; i < countStr.length(); i++) {
                chars[idx++] = countStr.charAt(i);
            }
        }
        return idx;
    }
}